### ISSUE TITLE
System Messages (/motion, /accel, /gyro) Sneak in and crash my GUI

### ISSUE THREAD

**Issue by JRSV** [monkeyswarm/MobMuPlatEditor#24]
I am having a weird problem when connecting mobmuplat to PD. I am directing messages according to their names, but somehow if the iPad moves it is able to insert a message like: /motion, /accel, /gyro, etc inside the stream that is only listening to other messages. I have tried [route] in various ways but still no luck. I cannot find a way to discriminate this messages. Can I disable this information somehow so it stops interfering and hanging PD's GUI? I am performing live tomorrow and this is a nasty surprise. 

any help will be appreciated.

thanks

**Comment by monkeyswarm** [comment]
When you say 'connect mobmuplat to pd', I'm assuming you mean opening a patch 'raw' (i.e. showing the patch, not a custom UI)

You should be able to filter them with [route]; you may need to use a [list trim] first in order to remove the 'list' token.

[list trim]
|
[route /motion /accel /gyro ]

and then use the last outlet of [route]. 
you can then pass it through [list] to make it a list again.

**Comment by JRSV** [comment]
No, I am referring to the Pd GUI that is receiving a message (in my debian machine). If the object, in this case a [* 127] receives the symbols mentioned (/motion, /accel, /tilts, /gyro, etc) sporadically it crashes the object and makes that controller un-responsive. I have managed to filter these annoying messages but I find it strange that they can just interrupt a stream that is not routed for them.

![Screenshot_2023-06-23_06-59-56](https://github.com/monkeyswarm/MobMuPlatEditor/assets/18346930/63870d31-d128-4a5b-8fde-dd643e644149)

**Comment by monkeyswarm** [comment]
I would suggest filtering them out on the mobile patch, to reduce network traffic (and the liklihood of an important message getting dropped, if using UDP).

I assume you have found a solution, closing issue.

**Comment by JRSV** [comment]
well.. before you close it, I do have to ask, why these messages can just enter the stream?? in my patch that goes with the template, the route object should be sufficient to discriminate them, however I still get them. Due to the urgency, I have fixed it this way, but if traffic improves if solved where you say, then I still cant fix it that way. I will try again post performance. Thanks

**Comment by monkeyswarm** [comment]
These messages enter your patch (on the phone) via 
[receive fromSystem]
If you don't want them, then don't connect that to anything.

See the list of messages in the section "System messages": https://danieliglesia.com/mobmuplat/doc/index.htm

If you _do_ want to use anything reported from the system, then it's up to you to filter that flood of system-provided messages (via [route]) to get the types of messages you want.

Feel free to post the patch on your phone which is failing to filter messages and I can take a look. If that's broken, then that's a Pd or libpd issue, not a MobMuPlat issue.
